### ISSUE TITLE
socket.timeout: timed out and websocket._exceptions.WebSocketTimeoutException: timed out

### ISSUE THREAD

**Issue by lml519** [oxwhirl/pymarl#33]
when i run the command "python3 src/main.py --config=qmix_smac --env-config=sc2 with env_args.map_name=2s3z".The output  comes "File "/usr/local/lib/python3.5/dist-packages/websocket/_socket.py", line 81, in recv
    bytes_ = sock.recv(bufsize)
  File "/usr/local/lib/python3.5/dist-packages/websocket/_socket.py", line 81, in recv
    bytes_ = sock.recv(bufsize)
  File "/usr/local/lib/python3.5/dist-packages/websocket/_socket.py", line 81, in recv
    bytes_ = sock.recv(bufsize)
  File "/usr/local/lib/python3.5/dist-packages/websocket/_socket.py", line 81, in recv
    bytes_ = sock.recv(bufsize)
  File "/usr/local/lib/python3.5/dist-packages/websocket/_socket.py", line 81, in recv
    bytes_ = sock.recv(bufsize)
socket.timeout: timed out
socket.timeout: timed out
socket.timeout: timed out
socket.timeout: timed out
socket.timeout: timed out
...
websocket._exceptions.WebSocketTimeoutException: timed out
websocket._exceptions.WebSocketTimeoutException: timed out
websocket._exceptions.WebSocketTimeoutException: timed out
"
Because of the length of the text, i copy a part of content, and i would like to send the whole component in the next comment.

**Comment by lml519** [comment]
now,the output changed into :
"
[ERROR 10:41:04] pymarl Failed after 0:00:51!
Traceback (most recent calls WITHOUT Sacred internals):
  File "/usr/lib/python3.5/contextlib.py", line 77, in __exit__
    self.gen.throw(type, value, traceback)
  File "src/main.py", line 37, in my_main
    run(_run, config, _log)
  File "/home/limenglin519/pymarl/src/run.py", line 48, in run
    run_sequential(args=args, logger=logger)
  File "/home/limenglin519/pymarl/src/run.py", line 179, in run_sequential
    learner.train(episode_sample, runner.t_env, episode)
  File "/home/limenglin519/pymarl/src/learners/q_learner.py", line 100, in train
    loss.backward()
  File "/usr/local/lib/python3.5/dist-packages/torch/tensor.py", line 150, in backward
    torch.autograd.backward(self, gradient, retain_graph, create_graph)
  File "/usr/local/lib/python3.5/dist-packages/torch/autograd/__init__.py", line 99, in backward
    allow_unreachable=True)  # allow_unreachable flag
RuntimeError: one of the variables needed for gradient computation has been modified by an inplace operation: [torch.cuda.FloatTensor [32, 68, 5, 11]], which is output 0 of SliceBackward, is at version 1; expected version 0 instead. Hint: enable anomaly detection to find the operation that failed to compute its gradient, with torch.autograd.set_detect_anomaly(True).

During handling of the above exception, another exception occurred:

Traceback (most recent calls WITHOUT Sacred internals):
  File "/usr/lib/python3.5/contextlib.py", line 77, in __exit__
    self.gen.throw(type, value, traceback)
  File "/usr/lib/python3.5/subprocess.py", line 1650, in wait
    raise TimeoutExpired(self.args, timeout)
subprocess.TimeoutExpired: Command '['tee', '-a', '/tmp/tmpj0067axl']' timed out after 1 seconds

"

**Comment by tabzraz** [comment]
It looks like an issue with the gradient computation in newer (>0.4) versions of pytorch.
I've just committed a change that should fix it.

**Comment by lml519** [comment]
@tabzraz thank u for ur help.I downloaded the package and rebuild it again. The problem have been solved. But i have another question now. How can i set the checkpoint path to load so that i can see the Name.replay . I set the default of Save_replay to TRUE. But after the end of the running,I can't find the Name.replay in the direction of replay in Starcraft2. I think the reason of this problem is  the checkpoint path . I want to know what path u set in default.yaml.

Thank u very much
